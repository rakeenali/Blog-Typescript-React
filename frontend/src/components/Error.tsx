import React from "react";
import styled from "styled-components";

interface IProps {
  message: string;
  closeError: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

const Error = (props: IProps) => {
  return (
    <Container>
      <CloseButton onClick={props.closeError}>&times;</CloseButton>
      <h2>{props.message}</h2>
    </Container>
  );
};

const Container = styled.div`
  &:before {
    content: "";
    position: fixed;
    top: 0;
    left: 0;
    overflow: hidden;
    height: 100vh;
    width: 100vw;
    background-color: rgba(0, 0, 0, 0.98);
    z-index: 100;
  }

  position: absolute;
  top: 0;
  left: 0;
  overflow: hidden;
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 500;

  h2 {
    color: red;
    background-color: #ff00003b;
    padding: 5px;
    width: 95%;
    margin: 0 auto;
    border-radius: 4px;
    text-align: center;
    font-size: 50px;
    z-index: 600;
  }
`;

const CloseButton = styled.button`
  position: absolute;
  top: 0;
  right: 5px;
  z-index: 100;
  font-size: 80px;
  border: none;
  color: red;
  background: none;
  cursor: pointer;

  &:hover {
    color: #ff0000bd;
  }
`;

export default Error;
