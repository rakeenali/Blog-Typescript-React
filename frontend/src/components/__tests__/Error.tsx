import '@testing-library/jest-dom/extend-expect'
import * as React from "react";
import { render, fireEvent } from "@testing-library/react";

import Error from "../Error";

describe("<Error/>", () => {

    test("render correctly", () => {
        const props = {
            message: "An error message",
            closeError: jest.fn()
        }
        const { container } = render(<Error {...props} />)
        expect(container.firstChild).toMatchSnapshot();
    })

    test("render the message and fire the close event", () => {
        const props = {
            message: "An error message",
            closeError: jest.fn()
        }

        const { queryByText, container } = render(<Error {...props} />)
        const closeButton = container.querySelector('button');

        expect(queryByText(/An error message/)).toBeInTheDocument()
        if (closeButton) {
            fireEvent.click(closeButton);
        }
        expect(props.closeError).toHaveBeenCalledTimes(1);
    });
})