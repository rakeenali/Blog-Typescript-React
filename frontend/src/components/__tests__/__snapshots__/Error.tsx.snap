// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error/> render correctly 1`] = `
<div
  class="sc-bdVaJa dAfcRe"
>
  <button
    class="sc-bwzfXH feCttl"
  >
    Ã—
  </button>
  <h2>
    An error message
  </h2>
</div>
`;
